src/main/scala/terminal/FileSystem.scala

```diff
@@ -0,0 +1,113 @@
+package terminal
+
+import java.util.NoSuchElementException
+
+
+class FileSystem {
+  //key = command;  value = how many paths need to be specified
+  val commandsAndNoOfPaths = Map("pwd" -> 0, "mkdir" -> 1, "ls" -> 0, "h" -> 0, "q" -> 0 )
+
+  var rootPath : Folders = _
+  var currentLevel : Folders = _
+  var cmdPrefix : String = ""
+  var path1_from : String = ""
+  var path2_to : String = ""
```

Comment:
In scala the preferred way to name variables is camelCase `path1From`. Also, these are not Filesystem attributes, more like command attributes

Reply:
Yes, these are equivalent to name1, name2 where these can be used like:
rm name1 name2
cp name1 name2
where name1 = from, name2 = to

I will still re-think this in a much elegant way. For the camelCase, yes very good point. 👍